server/database/migrations/2016_06_01_162617_comment_unification.php

```diff
@@ -0,0 +1,374 @@
+                            'created_at' => $commentMeta->created_at,
+                            'updated_at' => $commentMeta->updated_at,
+                        ]);
+                        break;
+                    case 'flag':
+                        $newFlag = AnnotationTypes\Flag::create();
+                        $this->updateTimestamps($newFlag, $commentMeta);
+                        DB::table('annotations')->insert([
+                            'user_id' => $commentMeta->user_id,
+                            'annotatable_id' => $commentAnnotationId,
+                            'annotatable_type' => Annotation::class,
+                            'annotation_type_id' => $newFlag->id,
+                            'annotation_type_type' => AnnotationTypes\Flag::class,
+                            'created_at' => $commentMeta->created_at,
+                            'updated_at' => $commentMeta->updated_at,
+                        ]);
+                        break;
+                    default:
+                        throw new Exception('Unknown comment meta value: "'.$commentMeta->meta_value.'"');
+                }
+            }
+        }
+
+        // AnnotationRange -> AnnotationTypes/Range
+        $annotationRanges = DB::table('annotation_ranges')->get();
+        foreach ($annotationRanges as $annotationRange) {
+            $newRange = AnnotationTypes\Range::create([
+                'start' => $annotationRange->start,
+                'end' => $annotationRange->end,
+                'start_offset' => $annotationRange->start_offset,
+                'end_offset' => $annotationRange->end_offset,
+            ]);
+
+            $this->updateTimestamps($newRange, $annotationRange);
+
+            $userId = DB
+                ::table('annotations')
+                ->where('id', $annotationRange->annotation_id)
+                ->value('user_id')
+                ;
+
+            DB::table('annotations')->insert([
+                'user_id' => $userId,
+                'annotatable_id' => $annotationRange->annotation_id,
+                'annotatable_type' => Annotation::class,
+                'annotation_type_id' => $newRange->id,
+                'annotation_type_type' => AnnotationTypes\Range::class,
+                'created_at' => $annotationRange->created_at,
+                'updated_at' => $annotationRange->updated_at,
+                'deleted_at' => $annotationRange->deleted_at,
+            ]);
+        }
+
+
+        // delete old places
+        Schema::table('annotations', function ($table) {
+            $table->dropColumn('quote');
+            $table->dropColumn('page');
+            $table->dropColumn('search_id');
+            $table->dropColumn('seen');
+            $table->dropColumn('text');
+            $table->dropColumn('uri');
+            $table->dropForeign(['doc_id']);
+            $table->dropColumn('doc_id');
+        });
+
+        Schema::drop('annotation_comments');
```

Comment:
Getting the following error when trying to run `make db-reset` and build from scratch.

`SQLSTATE[42S02]: Base table or view not found: 1146 Table 'madison.annotation_comments' doesn't exist (SQL: truncate `annotation_comments`)`